src/cicd/pipelines.ts

```diff
@@ -0,0 +1,321 @@
+      Parameters
+    ----------
+    stageId: str
+    Identifier of the stage
+    stage: Stage
+    Application stage instance
+    manualApprovals: Optional[bool]
+    Configure manual approvals. False by default
+      Returns
+    -------
+    pipeline : CICDPipelineStack
+    CICDPipelineStack
+    */
+    var manualApprovals = props.manualApprovals ?? false; // || this._config.get_env_config(stage_id).get('manual_approvals');
+
+    if (manualApprovals) {
+      this.pipeline?.addStage(props.stage, {
+        pre: [new ManualApprovalStep('PromoteTo' + toTitleCase(props.stageId))],
+      });
+    } else {
+      this.pipeline?.addStage(props.stage, {});
+    }
+
+    return this;
+  }
+
+  addSecurityLintStage(props: AddSecurityLintStageProps) {
+    /*
+    Add linting - cfn-nag, and bandit.
+      Parameters
+    ----------
+    stage_name: Optional[str]
+    Name of the stage
+    cloudAssemblyFileSet: Optional[IFileSetProducer]
+    Cloud assembly file set producer
+      Returns
+    -------
+    pipeline : CICDPipeline
+    CICD pipeline
+    */
+
+    var stageName = props.stageName ?? 'SecurityLint';
+    var cloudAssemblyFileSet =
+      props.cloudAssemblyFileSet ?? this.pipeline?.cloudAssemblyFileSet;
+
+    this.pipeline?.addWave(stageName, {
+      post: [
+        getCfnNagAction({
+          fileSetProducer: cloudAssemblyFileSet!,
+        }),
+        getBanditAction({
+          codePipelineSource: this.sourceAction!,
+        }),
+      ],
+    });
+
+    return this;
+  }
+
+  addTestStage(props: AddTestStageProps) {
+    /*
+    Add test - e.g. pytest.
+      Parameters
+    ----------
+    stage_name: Optional[str]
+    Name of the stage
+    cloudAssemblyFileSet: Optional[IFileSetProducer]
+    Cloud assembly file set
+    commands: Optional[List[str]]
+    Additional commands to run in the test. Defaults to './test.sh' otherwise
+      Returns
+    -------
+    pipeline : CICDPipelineStack
+    CICD pipeline
+    */
+    var stageName = props.stageName ?? 'Tests';
+    var cloudAssemblyFileSet =
+      props.cloudAssemblyFileSet ?? this.pipeline?.cloudAssemblyFileSet;
+    var commands = props.commands ?? ['./test.sh'];
+
+    this.pipeline?.addWave(stageName || 'Tests', {
+      post: [
+        getTestsAction({
+          fileSetProducer: cloudAssemblyFileSet!,
+          commands: commands,
+        }),
+      ],
+    });
+
+    return this;
+  }
+
+  addNotifications(props: AddNotificationsProps) {
+    /*
+    Add pipeline notifications. Create notification rule that sends events to the specified SNS topic.
+      Parameters
+    ----------
+    notificationRule: Optional[NotificationRule]
+    Override notification rule
+      Returns
+    -------
+    pipeline : CICDPipeline
+    CICD pipeline
+    */
+
+    this.notificationRule =
+      props.notificationRule ??
+      new NotificationRule(this, 'notification', {
```

Comment:
It would be good to capitalize ID Names, so that they are easier to parse when we view the resources in the CDK console.